test/ILLink.RoslynAnalyzer.Tests/TestChecker.cs

```diff
@@ -72,6 +71,12 @@ public override void VisitClassDeclaration (ClassDeclarationSyntax node)
 			CheckMember (node);
 		}
 
+		public override void VisitConstructorDeclaration (ConstructorDeclarationSyntax node)
```

Comment:
Fix for #2446

Reply:
Can you add a test for this? Something that Inside the static constructor calls a method annotated with RUC and just verify that we can use the ExpectedWarning attribute

Reply:
Good thing you mentioned this, I think I found a bug that the linker doesn't check in constructors for calls to methods with RUC. The test class for this is `WarningsInCtor` in RequiresCapability.cs. It calls a method annotated with RUC inside the constructors, but the linker doesn't seem to be producing a warning. I filed https://github.com/dotnet/linker/issues/2484 to track the issue.

Reply:
I think its because nothing references that code in the main method, since it's not referenced Linker will get rid of that piece of code